### PR TITLE
Nodepool nodes are labeled with nodepool id on AWS

### PR DESCRIPTION
**Pull request by fiunchinho**
This is a follow up of https://github.com/giantswarm/kubectl-gs/pull/481 since the label was missing from nodes on AWS.

[Our docs say](https://docs.giantswarm.io/advanced/node-pools/#assigning-workloads) that we use the label `giantswarm.io/machine-deployment` on AWS but `giantswarm.io/machine-pool` on Azure. This label can be anything that we want, so I'd suggest to use the same on all providers. This PR uses `giantswarm.io/machine-pool` as that's the value we are already using on Azure, and it makes explicit that we are using CAPI `MachinePools` under the hood. What do you think?

### CODE REVIEW

**Comment by AverageMarcus** [comment]
Big +1 for consistency!

Do we want to replace the `giantswarm.io/machine-deployment` completely? (e.g. here: https://github.com/giantswarm/kubectl-gs/blob/1e98bbfa29ecfe443432ca0f19eac4db1a402d10/internal/label/label.go#L10)

I think the `.golden` files will need updating too.

File: cmd/template/nodepool/provider/aws.go [hunk 1]
```diff
@@ -327,6 +328,10 @@ func newKubeadmConfigFromUnstructured(sshSSOPubKey string, sshdConfig string, o
 				Shell:  &shell,
 			},
 		}
+		kubeadmConfig.Spec.JoinConfiguration.NodeRegistration.KubeletExtraArgs = map[string]string{
+			"cloud-provider": "aws",
+			"node-labels":    fmt.Sprintf("giantswarm.io/machine-pool=%s", o.GetName()),
```
Comment (njuettner) [inline cmd/template/nodepool/provider/aws.go:333]: Do we want to add this label into apiextensions, it might be we need to pick it up elsewhere?

Comment (fiunchinho) [reply cmd/template/nodepool/provider/aws.go:333]: good call! Actually, it's already in there.

**Comment by fiunchinho** [comment]
> Big +1 for consistency!
> 
> Do we want to replace the `giantswarm.io/machine-deployment` completely? (e.g. here:
> 
> https://github.com/giantswarm/kubectl-gs/blob/1e98bbfa29ecfe443432ca0f19eac4db1a402d10/internal/label/label.go#L10
> 
> )
> I think the `.golden` files will need updating too.

That'd be a breaking change in our UX. I'd say let's change it when customers migrate to CAPI clusters.

**Review by njuettner — APPROVED** [review]
lgtm

**Review by anvddriesch — APPROVED** [review]

**Review by axbarsan — APPROVED** [review]

**Comment by marians** [comment]
@fiunchinho Thanks! Some questions:

- Are you also going to make the according docs changes?
- Is this working immediately for all releases?
- Will the old annotation `giantswarm.io/machine-deployment` still be set? Could be that customers have automation expecting it.

**Comment by fiunchinho** [comment]
> @fiunchinho Thanks! Some questions:
> 
> * Are you also going to make the according docs changes?
> * Is this working immediately for all releases?
> * Will the old annotation `giantswarm.io/machine-deployment` still be set? Could be that customers have automation expecting it.

I believe we don't have docs for CAPI clusters on GiantSwarm just yet. This is not changing how regular GiantSwarm clusters are templated with `kubectl-gs`, only `CAPA` clusters. So nothing will change for customers, as they are not using CAPA clusters yet.

**Pull request merged**